docs/source/1.0/spec/core/idl.rst

```diff
@@ -1438,6 +1445,97 @@ For example:
     string SensitiveText with [SensitiveString]
 
 
+.. _idl-target-elision:
+
+Target Elision
+--------------
+
+Having to completely redifine a resource identifier to use it in a structure or
+redefine a member from a mixin to add additional traits can be cumbersome and
+potentially error-prone. The type elision syntax can be used to cut down on
+that repetition by prefixing the member name with a ``$``. If a member is
+prefixed this way, its target will automatically be set to the target of a
+mixin member with the same name. The following example shows how to elide the
+target for a member inherited from a mixin:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    @mixin
+    structure IdBearer {
+        id: String
+    }
+
+    structure IdRequired {
+        @required
+        $id
+    }
+
+Additionally, structure shapes can reference a :ref:`resource <idl-resource>`
+shape to define members representing the resource's identifiers without having
+to redefine the target shape. In addition to prefixing a member with ``$``, the
+structure must also add ``from`` followed by the resource being referenced in
+the shape's definition before any mixins are specified. When using a resource
+reference, any member prefixed with ``$`` will first check for an identifier
+from that resource whose name matches the member name and use its target. If no
+identifier matches, mixins will then be checked. The following example shows a
+structure re-using an identifier definition from a resource:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    resource User {
+        identifiers: {
+            name: String
+            uuid: String
+        }
+    }
+
+    structure UserSummary from User {
+        $name
+        age: Short
+    }
+
+Note that the ``UserSummary`` structure does not attempt to define the
+``uuid`` identifier. When referencing a resource in this way, only the
+identifiers that are explicitly referenced are added to the structure. This
+allows structures to define subsets of identifiers, which can be useful for
+operations like create operations where some of those identifiers may be
+generated by the service.
+
+The target for a resource identifier can also be elided when the member's name
```

Comment:
> The current approach requires that traits are resolved before we can make sense of the resource binding (i.e., we need to know that resourceIdentifier is smithy.api#resourceIdentifier and not some other overloaded trait).

The resourceIdentifier trait will already have been resolved before we start looking at it, so this isn't adding any additional complexity. And we know it's `smithy.api#resourceIdentifier` because we're checking for that exact shape id.

The difference between having this feature and not having it is 3 lines of code, and omitting it makes it feel incomplete.

> We automatically inject a resourceIdentifier trait for the member

What is the benefit of doing this?

Reply:
Even if it’s a few lines, I think being able to use indirection to refer to properties of a resource adds a lot complexity for readers, potentially other tools implemented differently, and in IDEs that are trying to jump to the definition too.

Injecting resourceProperty traits and eventually property traits would be nice because “$” is adding these bindings explicitly through syntax, so rather than rely on implicit bindings later, we add the explicit traits. This makes it so that you can use selectors to look for these traits. Before doing this, I’d want to make sure this would work with resource properties when they ship though.

Reply:
This was resolved. What ended up being the associated change?

Reply:
You can see the rest int he most recent diff